tests/local_rma.cpp

```diff
@@ -0,0 +1,430 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+#include <gtest/gtest.h>
+#include <thread>
+#include <array>
+#include <gridtools/common/array.hpp>
+
+#include <ghex/threads/atomic/primitives.hpp>
+#include <ghex/threads/std_thread/primitives.hpp>
+#ifndef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+using threading = gridtools::ghex::threads::std_thread::primitives;
+#else
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+using threading = gridtools::ghex::threads::std_thread::primitives;
+#endif
+
+#include <ghex/bulk_communication_object.hpp>
+#include <ghex/structured/pattern.hpp>
+#include <ghex/structured/rma_range_generator.hpp>
+#include <ghex/structured/regular/domain_descriptor.hpp>
+#include <ghex/structured/regular/field_descriptor.hpp>
+#include <ghex/structured/regular/halo_generator.hpp>
+
+//#include <iomanip>
+//#include <future>
+//#ifdef __CUDACC__
+//#include <gridtools/common/cuda_util.hpp>
+//#include <gridtools/common/host_device.hpp>
+//#endif
+
+
+struct simulation_1
+{
+#ifdef __CUDACC__
+    template<typename T>
+    struct cuda_deleter
+    {
+        void operator()(T* ptr)
+        {
+            //cudaFree(ptr);
```

Comment:
problems? Cray MPI issues?

Reply:
yes, Cray stuff - I am working on this and will add a comment.